src/main/java/org/reactome/release/qa/diagram/DiagramDrugColorCheck.java

```diff
@@ -0,0 +1,112 @@
+                                "MostRecentAuthor");
+        return report;
+    }
+
+    private void checkPathwayDiagram(GKInstance diagram, DiagramGKBReader reader, QAReport report) throws Exception {
+        RenderablePathway pathway = reader.openDiagram(diagram);
+        String modDate = QACheckerHelper.getLastModificationAuthor(diagram);
+        List<Renderable> components = pathway.getComponents();
+        if (components == null || components.size() == 0)
+            return;
+        // For all rendered components.
+        for (Renderable component : components) {
+            // Check correctness of drug background.
+            List<Class<? extends Node>> classesToCheck = Arrays.asList(RenderableEntitySet.class,
+                                                                       RenderableComplex.class,
+                                                                       RenderableChemicalDrug.class,
+                                                                       RenderableProteinDrug.class,
+                                                                       RenderableRNADrug.class);
+            Class<Renderable> cls = (Class<Renderable>) component.getClass();
+            if (classesToCheck.contains(cls) && !hasCorrectColors(component)) {
+                // Add line to report.
+                report.addLine(diagram.getDBID().toString(),
+                               pathway.getDisplayName(),
+                               pathway.getReactomeDiagramId().toString(),
+                               component.getDisplayName(),
+                               component.getReactomeId().toString(),
+                               modDate);
+            }
+        }
+    }
+
+    /**
+     * Check if a component has the appropriate background color for its drug status.
+     *
+     * Returns true is color matches the status.
+     * Returns false if there is a mismatch error.
+     *
+     * @param component
+     * @return boolean
+     * @throws Exception
+     */
+    private boolean hasCorrectColors(Renderable component) throws Exception {
+        GKInstance instance = dba.fetchInstance(component.getReactomeId());
+        if (component.getReactomeId() == null || instance == null)
+            return true;
+
+        // Drug color check.
+        if (!InstanceUtilities.hasDrug(instance))
+            return true;
+
+        // Drug background and foreground check.
+        Object background = component.getBackgroundColor();
+        Object foreground = component.getForegroundColor();
+
+        // If background or foreground is null, default values will be used.
+        if (background == null || foreground == null)
+            return false;
+
+        boolean correctBackground = background.equals(DefaultRenderConstants.DEFAULT_DRUG_BACKGROUND);
+        boolean correctForeground = foreground.equals(DefaultRenderConstants.DEFAULT_DRUG_FOREGROUND);
+
+        if (correctBackground && correctForeground)
```

Comment:
Merge all the following statements as:

return correctBackground && correctForeground;

Reply:
Commit [4765e01](https://github.com/reactome/release-qa/pull/20/commits/4765e0111f6455a8ce5a0118935d7c98095685ca) fixes all code issues here and above.